monkey/monkey_island/cc/services/attack/technique_reports/pba_technique.py

```diff
@@ -0,0 +1,41 @@
+import abc
+
+from monkey_island.cc.services.attack.attack_config import AttackConfig
+from monkey_island.cc.database import mongo
+from common.utils.attack_utils import ScanStatus
+from monkey_island.cc.services.attack.technique_reports import AttackTechnique
+
+
+class PostBreachTechnique(AttackTechnique, metaclass=abc.ABCMeta):
+    @property
+    @abc.abstractmethod
+    def pba_names(self):
+        """
+        :return: name of post breach action
+        """
+        pass
+
+    @classmethod
+    def get_pba_query(cls, post_breach_action_names):
+        return [{'$match': {'telem_category': 'post_breach',
+                            '$or': [{'data.name': pba_name} for pba_name in post_breach_action_names]}},
+                {'$project': {'_id': 0,
+                              'machine': {'hostname': '$data.hostname',
+                                          'ips': ['$data.ip']},
+                              'result': '$data.result'}}]
```

Comment:
I'd prefer for us to use mongoengine models and not direct Mongo queries. How long do you think that will take? Where will this impact?

Reply:
While using mongoengine will define a schema for the data and make development easier, is it worth doing considering that it'll definitely be slower than pymongo (see [this](https://stackoverflow.com/a/35274930/10629482) and [this](https://stackoverflow.com/a/41323384/10629482))?

<!---
https://stackoverflow.com/questions/62435850/mongodb-to-run-faster-queries-which-is-better-pymongo-or-mongoengine

 --->